dymos/examples/balanced_field/test/test_balanced_field_length.py

```diff
@@ -0,0 +1,452 @@
+        p.set_val('traj.rotate.states:v', rotate.interpolate(ys=[80, 85.0], nodes='state_input'))
+        p.set_val('traj.rotate.polynomial_controls:alpha', 0.0, units='deg')
+
+        p.set_val('traj.climb.t_initial', 30)
+        p.set_val('traj.climb.t_duration', 20)
+        p.set_val('traj.climb.states:r', climb.interpolate(ys=[5000, 5500.0], nodes='state_input'), units='ft')
+        p.set_val('traj.climb.states:v', climb.interpolate(ys=[160, 170.0], nodes='state_input'), units='kn')
+        p.set_val('traj.climb.states:h', climb.interpolate(ys=[0, 35.0], nodes='state_input'), units='ft')
+        p.set_val('traj.climb.states:gam', climb.interpolate(ys=[0, 5.0], nodes='state_input'), units='deg')
+        p.set_val('traj.climb.controls:alpha', 5.0, units='deg')
+
+        return p
+
+    def test_restart_from_sol(self):
+        p = self._make_problem()
+        dm.run_problem(p, run_driver=True, simulate=False)
+
+        sol_results = om.CaseReader('dymos_solution.db').get_case('final')
+
+        assert_near_equal(sol_results.get_val('traj.climb.timeseries.states:r')[-1], 2016, tolerance=0.01)
+
+        dm.run_problem(p, run_driver=True, simulate=True, restart='dymos_solution.db')
+
+        sol_results = om.CaseReader('dymos_solution.db').get_case('final')
+        sim_results = om.CaseReader('dymos_simulation.db').get_case('final')
+
+        assert_near_equal(sol_results.get_val('traj.climb.timeseries.states:r')[-1], 2016, tolerance=0.01)
+        assert_near_equal(sim_results.get_val('traj.climb.timeseries.states:r')[-1], 2016, tolerance=0.01)
+
+        assert_near_equal(sol_results.get_val('traj.rto.timeseries.states:r')[-1], 2016, tolerance=0.01)
+        assert_near_equal(sim_results.get_val('traj.rto.timeseries.states:r')[-1], 2016, tolerance=0.01)
+
+    def test_restart_from_sim(self):
+        p = self._make_problem()
+        dm.run_problem(p, run_driver=True, simulate=True)
+
+        sol_results = om.CaseReader('dymos_solution.db').get_case('final')
+
+        assert_near_equal(sol_results.get_val('traj.climb.timeseries.states:r')[-1], 2016,
+                          tolerance=0.01)
+
+        dm.run_problem(p, run_driver=True, simulate=True, restart='dymos_simulation.db')
+
+        sol_results = om.CaseReader('dymos_solution.db').get_case('final')
+        sim_results = om.CaseReader('dymos_simulation.db').get_case('final')
+
+        assert_near_equal(sol_results.get_val('traj.climb.timeseries.states:r')[-1], 2016,
+                          tolerance=0.01)
+        assert_near_equal(sim_results.get_val('traj.climb.timeseries.states:r')[-1], 2016,
+                          tolerance=0.01)
+
+        assert_near_equal(sol_results.get_val('traj.rto.timeseries.states:r')[-1], 2016,
+                          tolerance=0.01)
+        assert_near_equal(sim_results.get_val('traj.rto.timeseries.states:r')[-1], 2016,
+                          tolerance=0.01)
+
+
```

Comment:
The "default values" case is an experiment to play with setting all values via the `add_state`, `add_control`, `set_time_options`, etc methods.  When playing around with adding and removing phases, having the phase definition and the values setting in two parts of the file is a bit annoying.  The downside of this method is that the values have to be set with their problem units, unlike using the `set_val` method.

Reply:
This makes those methods more consistent with all the other openmdao methods too.